harness/determined/pytorch/_pytorch_context.py

```diff
@@ -82,6 +82,12 @@ def __init__(self, *args: Any, **kwargs: Any) -> None:
         self._reducers = pytorch._PyTorchReducerContext()
         self._determined_profiler = None  # type: Optional[profiler.ProfilerAgent]
 
+        optimizations_config = self.get_optimizations_config()
+        self.aggregation_frequency = cast(int, optimizations_config.get("aggregation_frequency"))
+        self.fp16_compression = cast(bool, optimizations_config.get("gradient_compression"))
+        self.average_aggregated_gradients = cast(bool, optimizations_config.get("average_aggregated_gradients"))
+        self.average_training_metrics = cast(bool, optimizations_config.get("average_training_metrics"))
```

Comment:
Do you mean for these to be user-facing APIs?  If not they need to each have an underscore prefix.

There are some old attributes that didn't get this treatment but also aren't really public APIs, like context.env and context.experiment_config, but we should be consistent with new values.